doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+
+        To clear the model, a new estimator should be constructed, for instance
+        with :func:`base.clone`.
+
+    ``predict``
+        Makes a prediction for each sample. In a :term:`classifier` or
+        :term:`regressor`, this prediction is in the same target space used in
+        fitting (e.g. one of {'red', 'amber', 'green'} if the `y` in fitting
+        consisted of these strings).  In a :term:`clusterer` or :term:`outlier
+        detector` the prediction is an integer.
+
+        TODO
+
+        Return type is array ...
+
+        classifier
+            TODO
+
+        clusterer
+            TODO
+            -1 in DBSCAN
+
+        outlier detector
+            TODO
+
+        regressor
+
+        Mention ``return_std``
+
+    ``predict_log_proba``
+        TODO
+
+    ``predict_proba``
+        TODO
+
+        Output conventions are like those for ``decision_function`` except
+        in the :term:`binary` case.
+
+        Like other methods, ``predict_proba`` should only be present when the
+        estimator can make probabilistic predictions.  In some cases, this
+        means that the method will only appear after fitting.
+
+    ``score``
+        TODO
+
+    ``score_samples``
+        TODO
+
+    ``set_params``
+        TODO
+
+    ``split``
+        TODO
+
+.. _glossary_parameters:
+
+Parameters
+==========
+
+These common parameter names, specifically used in estimator construction
+(see concept :term:`parameter`) sometimes also appear as function and
+non-estimator parameters with similar semantics.
+
+.. glossary::
+
+    ``affinity``
+        TODO
+        See also :term:`kernel`
+
+        XXX: hierarchical clustering uses ``affinity`` for a distance metric.
+
+    ``class_weight``
+        Used to specify sample weights when fitting classifiers as a function
+        of the :term:`target` class.  Where the :term:`sample_weight`
+        :term:`sample property` is also supported and given, it is multiplied
+        by the ``class_weight`` contribution. Similarly, where ``class_weight``
+        is used in a :term:`multioutput` (including :term:`multilabel`) tasks,
+        the weights are multiplied across outputs (i.e. columns of ``y``).
+
+        By default all samples have equal weight such that classes are
+        effectively weighted by their their prevalence in the training data.
+        This could be achieved explicitly with ``class_weight={label1: 1,
+        label2: 1, ...}`` for all class labels.
+
+        More generally, ``class_weight`` is specified as a dict mapping class
+        labels to weights (``{class_label: weight}``), such that each sample
+        of the named class is given that weight.
+
+        ``class_weight='balanced'`` can be used to give all classes
+        equal weight by giving each sample a weight inversely related
+        to its class's prevalence in the training data:
+        ``n_samples / (n_classes * np.bincount(y))``.
```

Comment:
do we guarantee that this is the same as replicating the samples? I think we do, right?

Reply:
Well, that's true of `sample_weight` in general. It's more like asymptotic oversampling...